### PR TITLE
More accessible screen layout settings.

### PR DESCRIPTION
**Pull request by Cereal-Killa**
Depending on the game I'm playing I find myself needing to switch screen layouts very often and it's currently a hassle to have to open the settings dialog in order to do that so I've added shortcuts for the screen layout options and swap screen option on the main menu plus I added a keyboard shortcut to quickly toggle between the available layouts during game play (F10). I've also updated the swap screen shortcut (F9 now, used to be Ctrl + Tab) so it's next to the layout toggle shortcut by default (F9 and F10).

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3340)

### CODE REVIEW

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -262,8 +270,9 @@ void GMainWindow::InitializeRecentFileMenuActions() {
 
 void GMainWindow::InitializeHotkeys() {
     RegisterHotkey("Main Window", "Load File", QKeySequence::Open);
-    RegisterHotkey("Main Window", "Swap Screens", QKeySequence::NextChild);
     RegisterHotkey("Main Window", "Start Emulation");
+    RegisterHotkey("Main Window", "Swap Screens", QKeySequence(tr("F9"))); <--
```
Comment (Cereal-Killa) [inline src/citra_qt/main.cpp]: Forgot to remove the " <-" at the end of the line.

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -185,7 +185,15 @@ void GMainWindow::InitializeWidgets() {
         statusBar()->addPermanentWidget(label, 0);
     }
     statusBar()->setVisible(true);
+
+    // Removes an ugly inner border from the status bar widgets under Linux
```
Comment (Cereal-Killa) [inline src/citra_qt/main.cpp]: This is an old push I've made which lacked an explanation as to why this "hack" was added.

File: src/citra_qt/main.cpp [hunk 3]
```diff
@@ -262,8 +270,9 @@ void GMainWindow::InitializeRecentFileMenuActions() {
 
 void GMainWindow::InitializeHotkeys() {
     RegisterHotkey("Main Window", "Load File", QKeySequence::Open);
-    RegisterHotkey("Main Window", "Swap Screens", QKeySequence::NextChild);
     RegisterHotkey("Main Window", "Start Emulation");
+    RegisterHotkey("Main Window", "Swap Screens", QKeySequence(tr("F9"))); <--
+    RegisterHotkey("Main Window", "Toggle Screen Layout", QKeySequence(tr("10")))<--;
```
Comment (Cereal-Killa) [inline src/citra_qt/main.cpp]: Again the "<--" needs to be cleaned.

File: src/citra_qt/main.cpp [hunk 4]
```diff
@@ -262,8 +270,9 @@ void GMainWindow::InitializeRecentFileMenuActions() {
 
 void GMainWindow::InitializeHotkeys() {
     RegisterHotkey("Main Window", "Load File", QKeySequence::Open);
-    RegisterHotkey("Main Window", "Swap Screens", QKeySequence::NextChild);
```
Comment (Cereal-Killa) [inline src/citra_qt/main.cpp]: I thought that this function should have a shortcut close to the screen layout toggle and full screen toggle shortcuts.

**Comment by Cereal-Killa** [comment]
The only commit that I intended to push was 45fe8ff, I'm still getting a hang of Git, sorry about that.

**Comment by Cereal-Killa** [comment]
"A picture is worth a thousand words":

citra-qt_2017-12-31_15-09-56

**Comment by Cereal-Killa** [comment]
I just noticed as well that on Canary the disassembler key bindings do conflict with the ones I've set here:

image

**Comment by Cereal-Killa** [comment]
Still getting "Ambiguous shortcut overload" warning in the console when pressing F9 in windowed mode, shortcut still works though, in full screen mode the warning goes away.

File: src/citra_qt/main.h [hunk 5]
```diff
@@ -48,6 +48,7 @@ class GMainWindow : public QMainWindow {
 public:
     void filterBarSetChecked(bool state);
     void UpdateUITheme();
+    void SyncMenuUISettings();
```
Comment (lioncash) [inline src/citra_qt/main.h]: This should likely be made private. This is something the signal/slot mechanism should handle if it's ever needed outside the main window class.

**Comment by EzekielRage** [comment]
Can you add remapping to a Controller? Would be great to switch screens with the press of a button? Maybe the Home button or the right thumbstick or something^^

File: src/citra_qt/main.h (hunk 5 above)
Comment (Cereal-Killa) [reply src/citra_qt/main.h]: All set.

**Comment by Cereal-Killa** [comment]
As of remapping those functions to the controller I think it totally makes sense, I'm not sure if I should do a separate pull request for that.

**Comment by EzekielRage** [comment]
I asked on the discord yesterday and from what I understand (could be wrong) its oaky if you put it in this one, too. That's why tehy told me to comment on this^^

**Comment by Cereal-Killa** [comment]
Tks, sounds good then, let me get started here.

**Comment by Cereal-Killa** [comment]
image

**Comment by Cereal-Killa** [comment]
True, plus after struggling to figure out how to hook it up I asked @wwylele for some help and explained to me that it could be challenging considering the current code was designed for real 3DS controls, he suggested reimplementing the QT controller interface and I agreed with it.

That being said a "Shortcut" tab combined with a separate QT key handling mechanism for these non-3DS functions should be definitely the way to go, I will see if I can do some work on it this week.

Thank you.

**Comment by baka0815** [comment]
I think (personal opinion) that this should happen in a separate PR.
Having the fixed hot key, removing the last commit (controller binding) this PR should be good to go.
Rebuilding the GUI with a "Shortcut" tab can then be done in a separate PR.

But I'm not part of the team, so just personal opinion here.

**Comment by BreadFish64** [comment]
I agree with @baka0815, Ideally, the whole hotkey system needs to be overhauled.

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2018-01-09T16:19:30Z: https://github.com/citra-emu/citra/compare/743c247f46c5fea2da7caf05dd3f9f102e91f99d...Cereal-Killa:9324de714ce5ed404e2485c68bce451abedc8563

**Comment by Cereal-Killa** [comment]
Agreed, just removed the controller binding changes.

**Comment by Cereal-Killa** [comment]
Let me know if this is looking alright or needs any adjustment, thanks.

**Comment by FearlessTobi** [comment]
You should fix your Clang-format.

File: src/citra_qt/main.cpp [hunk 6]
```diff
@@ -869,12 +894,56 @@ void GMainWindow::ToggleWindowMode() {
     }
 }
 
+void GMainWindow::ChangeScreenLayout() {
+    Settings::LayoutOption newlayout = Settings::LayoutOption::Default;
+
+    if (ui.action_Screen_Layout_Default->isChecked())
```
Comment (jroweboy) [inline src/citra_qt/main.cpp]: Nitpick: Prefer `{}` around these

File: src/citra_qt/main.cpp [hunk 7]
```diff
@@ -869,12 +894,56 @@ void GMainWindow::ToggleWindowMode() {
     }
 }
 
+void GMainWindow::ChangeScreenLayout() {
+    Settings::LayoutOption newlayout = Settings::LayoutOption::Default;
```
Comment (jroweboy) [inline src/citra_qt/main.cpp]: style nitpick: `new_layout` instead

File: src/citra_qt/main.cpp [hunk 8]
```diff
@@ -869,12 +894,56 @@ void GMainWindow::ToggleWindowMode() {
...
+        newlayout = Settings::LayoutOption::LargeScreen;
+    else if (ui.action_Screen_Layout_Side_by_Side->isChecked())
+        newlayout = Settings::LayoutOption::SideScreen;
+
+    Settings::values.layout_option = newlayout;
+    Settings::Apply();
+}
+
+void GMainWindow::ToggleScreenLayout() {
+    Settings::LayoutOption newlayout = Settings::LayoutOption::Default;
```
Comment (jroweboy) [inline src/citra_qt/main.cpp]: ditto

File: src/citra_qt/main.cpp [hunk 9]
```diff
@@ -869,12 +894,56 @@ void GMainWindow::ToggleWindowMode() {
...
+    case Settings::LayoutOption::SideScreen:
+        newlayout = Settings::LayoutOption::Default;
+        break;
+    }
+
+    Settings::values.layout_option = newlayout;
+    Settings::Apply();
+}
+
+void GMainWindow::SwapScreens() {
```
Comment (jroweboy) [inline src/citra_qt/main.cpp]: Eh. I don't like that we renamed the `slot` to not have `On` in front of the name. But renaming all those to a common name should be a seperate pr

**Comment by EzekielRage** [comment]
is there a pull request for the new controler bindings (swap screeen) yet? :)

**Comment by Cereal-Killa** [comment]
Thanks @jroweboy, let me know if it looks ok now.
@EzekielRage Not yet...

**Comment by jroweboy** [comment]
@Cereal-Killa there is a merge conflict thats preventing this from being merged. If you would be so kind as to rebase off master, and squash some of the extra commits so we can get this merged. Lemme know if you are struggling to do this, and I'll be glad to help out.

**Comment by Cereal-Killa** [comment]
@jroweboy Yes I do need some help with that. I'll be on Discord whenever it's good for you, I appreciate it.

**Pull request merged**